backends/guest/common/util.c

```diff
@@ -38,6 +38,37 @@ const struct signature_type_info signature_type_list[] = {
 static uint8_t append[] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01 };
 static uint8_t replace[] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
 
+/*
+ * check it whether given variable is trustedcadb
+ */
+bool is_trustedcadb_variable(const char *variable_name)
+{
+	int len = strlen(variable_name);
+
+	if (memcmp(variable_name, TRUSTEDCADB_VARIABLE, len) == 0)
+		return true;
+
+	return false;
+}
+
+void print_timestamp(timestamp_t t)
+{
+	printf("%04d-%02d-%02d %02d:%02d:%02d UTC\n", t.year, t.month, t.day, t.hour, t.minute,
+	       t.second);
+}
+
+void read_timestamp(const uint8_t *esl_data)
+{
+	timestamp_t timestamp;
+
+	if (esl_data == NULL)
+		return;
+
+	memcpy(&timestamp, esl_data + 1, TIMESTAMP_LEN - 1);
```

Comment:
Why do we skip the first byte of the timestamp here?

Also, since this function does not do any check on the size of the buffer `*esl_data` points to, please include a comment that states that the caller must confirm the input buffer contains at least `TIMESTAMP_LEN` bytes.

Reply:
From the structure of the plpks_get_variable under linux/arch/powerpc/platforms/pseries/plpks-secvar.c, it seems the esl_data size will always be fixed to align with timestamp_len, my concern is that do we have plans to support variable key sizes in the future? In such a case this will need refactoring...

Reply:
This doesn't warrant a comment as per my understanding because we have used TIMESTAMP_LEN at multiple places and I don't think there can be a scenario where we will get esl_data_size less than the TIMESTAMP_LEN.